descartes_core/include/descartes_core/robot_model.h

```diff
@@ -107,7 +114,8 @@ class RobotModel
    * a fixed location relative to the last link in 'group_name'.
    */
   virtual bool initialize(const std::string &robot_description, const std::string &group_name,
-                          const std::string &world_frame, const std::string &tcp_frame) = 0;
+                          const std::string &world_frame, const std::string &tcp_frame,
+                          planning_scene_monitor::PlanningSceneMonitorPtr psm = nullptr) = 0;
```

Comment:
We went to great lengths to keep Descartes independent of MoveIt.  I believe that this is still true and at least some of the tests exercise Descartes independent of MoveIt, though it has been awhile.  

My suggested approach is to create a Descartes interface class and then have one of the implementations be the MoveIt planning scene OR why not just include the planning scene inside the `ikfast_moveit_state_adapter.h`?  While the robot model primarily performs kinematics, it's intent was to having collision checking embedded (otherwise it would have been called robot kinematics).

Reply:
What I've done for projects is create a function specific to `MoveItStateAdapter` and its children that looks like: `PlanningScene& getPlanningScene()`. This allows users to mutate the planning scene provided they know their robot is descended from the MoveIt adapter line of Descartes robot models.

What would that Descartes interface look like? Would it be our generic representation around the collision checking? E.g.
```c++

class CollisionInterface {
public:
  virtual bool isInCollision(const std::vector<double>& joints) const = 0;
}
```

And we would inject that into child classes?